apps/io-services-cms-webapp/src/watchers/on-service-lifecycle-change.ts

```diff
@@ -65,58 +65,66 @@ const onApproveHandler =
         max_allowed_payment_amount: MAX_ALLOWED_PAYMENT_AMOUNT,
       },
       autoPublish: ServiceLifecycle.getAutoPublish(item),
-      kind: "RequestPublicationItem",
     },
   });
 
 const onDeleteHandler = (
   item: ServiceLifecycle.ItemType
-): RequestUnpublicationItem => ({
-  requestPublication: {
+): RequestDeletionItem => ({
+  requestDeletion: {
     id: item.id,
-    kind: "RequestUnpublicationItem",
   },
 });
 
-// FIXME: fix request historicization action (avoid to call onAnyChangesHandler foreach case)
+const getSpecificAction =
+  (config: IConfig) =>
+  (
+    item: ServiceLifecycle.ItemType
+  ): O.Option<OnSubmitActions | OnApproveActions | OnDeleteActions> => {
+    switch (item.fsm.state) {
+      case "submitted":
+        return pipe(item, onSubmitHandler, O.some);
+      case "approved":
+        return pipe(item, onApproveHandler(config), O.some);
+      case "deleted":
+        return pipe(item, onDeleteHandler, O.some);
+      default:
+        return O.none;
+    }
+  };
+
 export const handler =
   (
     config: IConfig
-  ): RTE.ReaderTaskEither<
+  ): RE.ReaderEither<
     { item: ServiceLifecycle.ItemType },
     Error,
     | RequestHistoricizationAction
     | ((OnSubmitActions | OnApproveActions | OnDeleteActions) &
         RequestHistoricizationAction)
   > =>
-  ({ item }) => {
-    if (item.fsm.lastTransition !== SYNC_FROM_LEGACY) {
-      switch (item.fsm.state) {
-        case "submitted":
-          return pipe(
-            item,
-            onSubmitHandler,
-            (actions) => ({ ...actions, ...onAnyChangesHandler(item) }),
-            TE.right
-          );
-        case "approved":
-          return pipe(
-            item,
-            onApproveHandler(config),
-            (actions) => ({ ...actions, ...onAnyChangesHandler(item) }),
-            TE.right
-          );
-        case "deleted":
-          return pipe(
-            item,
-            onDeleteHandler,
-            (actions) => ({ ...actions, ...onAnyChangesHandler(item) }),
-            TE.right
-          );
-        default:
-          return TE.right(onAnyChangesHandler(item));
-      }
-    } else {
-      return TE.right(onAnyChangesHandler(item));
-    }
-  };
+  ({ item }) =>
+    pipe(
+      item,
+      onAnyChangesHandler,
+      E.right,
+      E.chain((historicizationAction) =>
+        pipe(
+          item.fsm.lastTransition !== SYNC_FROM_LEGACY ||
```

Comment:
can you add a comment?

Reply:
2d32e4c8978af5f93573a9c52ff4768d6692b4cc